### PR TITLE
Allow CPU build without TBB.

### PR DESCRIPTION
**Pull request by silee2**
Standalone usage sometimes require small binary size. Providing an option to build CPU backend without TBB can help reducing size.

### CODE REVIEW

File: src/ngraph/runtime/cpu/CMakeLists.txt [hunk 1]
```diff
@@ -195,7 +200,10 @@ if (NGRAPH_CPU_ENABLE)
     endif()
 
     add_dependencies(cpu_backend ext_mkldnn ext_eigen)
-    target_link_libraries(cpu_backend PUBLIC ngraph libmkldnn libeigen libjson libtbb)
+    target_link_libraries(cpu_backend PUBLIC ngraph libmkldnn libeigen libjson)
+    if (NGRAPH_TBB_ENABLE)
+        target_link_libraries(cpu_backend PUBLIC libtbb)
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/CMakeLists.txt]: I think this should be PRIVATE. Probably should make libmkldnn, libeigen and libjson PRIVATE too. codegen too.

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 2]
```diff
@@ -457,6 +461,7 @@ void runtime::cpu::CPU_ExternalFunction::compile()
     codegen::CodeWriter writer;
 
     writer << "// Generated by the nGraph CPU backend\n";
+#if defined(NGRAPH_TBB_ENABLE)
```
Comment (rkimballn1) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Can we remove the runtime enable/disable for TBB? We probably don't need both.

File: src/ngraph/runtime/cpu/CMakeLists.txt (hunk 1 above)
Comment (silee2) [reply src/ngraph/runtime/cpu/CMakeLists.txt]: Had to make it PUBLIC again as it breaks compiling a unit test.

**Comment by silee2** [comment]
Closing since TBB removal is not required yet.

**Pull request merged**